cmd/neofs-adm/internal/modules/morph/subnet.go

```diff
@@ -910,3 +802,405 @@ func init() {
+		},
+	}, nil
+}
+
+func notaryAccounts(alphabet, acc *wallet.Account) []*wallet.Account {
+	aa := make([]*wallet.Account, 0, 4)
+
+	// proxy
+	aa = append(aa, &wallet.Account{
+		Contract: &wallet.Contract{
+			Deployed: true,
+		},
+	})
+
+	// alphabet
+	aa = append(aa, alphabet)
+
+	// caller's account
+	aa = append(aa, acc)
+
+	// last one is a placeholder for notary contract account
+	aa = append(aa, &wallet.Account{
+		Contract: &wallet.Contract{},
+	})
+
+	return aa
+}
+
+func notaryWitnesses(c Client, alphabet, acc *wallet.Account, tx *transaction.Transaction) []transaction.Witness {
+	ww := make([]transaction.Witness, 0, 4)
+
+	// empty proxy contract witness
+	ww = append(ww, transaction.Witness{
+		InvocationScript:   []byte{},
+		VerificationScript: []byte{},
+	})
+
+	// alphabet multi-address witness
+	ww = append(ww, transaction.Witness{
+		InvocationScript: append(
+			[]byte{byte(opcode.PUSHDATA1), 64},
+			make([]byte, 64)...,
+		),
+		VerificationScript: alphabet.GetVerificationScript(),
+	})
+
+	magicNumber, _ := c.GetNetwork()
+
+	// caller's witness
+	ww = append(ww, transaction.Witness{
+		InvocationScript: append(
+			[]byte{byte(opcode.PUSHDATA1), 64},
+			acc.PrivateKey().SignHashable(uint32(magicNumber), tx)...),
+		VerificationScript: acc.GetVerificationScript(),
+	})
+
+	// notary contract witness
+	ww = append(ww, transaction.Witness{
+		InvocationScript: append(
+			[]byte{byte(opcode.PUSHDATA1), 64},
+			make([]byte, 64)...,
+		),
+		VerificationScript: []byte{},
+	})
+
+	return ww
+}
+
+func invocationParams(args ...interface{}) ([]smartcontract.Parameter, error) {
+	params := make([]smartcontract.Parameter, 0, len(args))
+
+	for i := range args {
+		param, err := toStackParameter(args[i])
+		if err != nil {
+			return nil, err
+		}
+
+		params = append(params, param)
+	}
+
+	return params, nil
+}
+
+func toStackParameter(value interface{}) (smartcontract.Parameter, error) {
+	var result = smartcontract.Parameter{
+		Value: value,
+	}
+
+	switch v := value.(type) {
+	case []byte:
+		result.Type = smartcontract.ByteArrayType
+	case int:
+		result.Type = smartcontract.IntegerType
+		result.Value = int64(v)
+	case int64:
+		result.Type = smartcontract.IntegerType
+		result.Value = v
+	case uint64:
+		result.Type = smartcontract.IntegerType
+		result.Value = int64(v)
+	case [][]byte:
+		arr := make([]smartcontract.Parameter, 0, len(v))
+		for i := range v {
+			elem, err := toStackParameter(v[i])
+			if err != nil {
+				return result, err
+			}
+
+			arr = append(arr, elem)
+		}
+
+		result.Type = smartcontract.ArrayType
+		result.Value = arr
+	case string:
+		result.Type = smartcontract.StringType
+	case util.Uint160:
+		result.Type = smartcontract.ByteArrayType
+		result.Value = v.BytesBE()
+	case noderoles.Role:
+		result.Type = smartcontract.IntegerType
+		result.Value = big.NewInt(int64(v))
+	case keys.PublicKeys:
+		arr := make([][]byte, 0, len(v))
+		for i := range v {
+			arr = append(arr, v[i].Bytes())
+		}
```

Comment:
All of this is not needed if you use `invokeFunction`, which accepts `[]interface{}`.